server/src/main/java/org/opensearch/index/compositeindex/datacube/startree/aggregators/MetricStatFieldPair.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.index.compositeindex.datacube.startree.aggregators;
+
+import org.opensearch.index.compositeindex.datacube.MetricStat;
+
+import java.util.Comparator;
+
+/**
+ * Builds aggregation function and doc values field pair to support various aggregations
+ * @opensearch.experimental
+ */
+public class MetricStatFieldPair implements Comparable<MetricStatFieldPair> {
```

Comment:
I think we can go with complex object instead of going this route, as suggested in base single tree builder.